### PR TITLE
[ST] test availability during nodes drain

### PR DESCRIPTION
**Pull request by kornys**
Signed-off-by: David Kornel <kornys@outlook.com>

### Type of change

_Select the type of your PR_

- Test

### Description

Added test for check strimzi behavior during kubernetes node drain.

### CODE REVIEW

File: systemtest/src/test/java/io/strimzi/systemtest/specific/ClusterOperationST.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        KafkaResource.kafkaPersistent(CLUSTER_NAME, 3, 3)
+                .editOrNewSpec()
+                .editEntityOperator()
+                .editUserOperator()
+                .withReconciliationIntervalSeconds(30)
+                .endUserOperator()
+                .endEntityOperator()
+                .endSpec()
+                .done();
+
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/specific/ClusterOperationST.java:59]: We use indentation across whole project for piece of code like this. You should follow our current practice.

File: systemtest/src/test/java/io/strimzi/systemtest/specific/ClusterOperationST.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        topicNames.forEach(topicName -> KafkaTopicResource.topic(CLUSTER_NAME, topicName, 3, 3, 2).done());
+
+        String producerAdditionConfiguration = "delivery.timeout.ms=20000\nrequest.timeout.ms=20000";
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/specific/ClusterOperationST.java:62]: Just a note: this is used to kinda simulate worse environment for clients cause we already had issues, that tests were green, but users were able to hit issues with producers and lost messages.

File: systemtest/src/test/java/io/strimzi/systemtest/specific/ClusterOperationST.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        consumerNames.forEach(consumerName -> kubeClient().getClient().batch().jobs().inNamespace(NAMESPACE).withName(consumerName).delete());
+    }
+
+    @BeforeAll
+    void setup() {
+        ResourceManager.setClassResources();
+        prepareEnvForOperator(NAMESPACE);
+
+        applyRoleBindings(NAMESPACE);
+        BundleResource.clusterOperator(NAMESPACE).done();
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/specific/ClusterOperationST.java:96]: Does it means we cannot use OLM/Helm install for this scenario?

File: systemtest/src/main/java/io/strimzi/systemtest/annotations/RequiredMinKubeApiVersion.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target({ElementType.METHOD, ElementType.TYPE})
+@Retention(RetentionPolicy.RUNTIME)
+@ExtendWith(RequiredMinKubeApiVersionCondition.class)
+public @interface RequiredMinKubeApiVersion {
+    double version() default 0;
```
[the remark this answers is not in the document]
Comment (kornys) [reply systemtest/src/main/java/io/strimzi/systemtest/annotations/RequiredMinKubeApiVersion.java:18]: impossible, you have to use constant value in annotation, so 0 is the best value I guess

File: systemtest/src/main/java/io/strimzi/systemtest/annotations/RequiredMinKubeApiVersionCondition.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    @Override
+    public ConditionEvaluationResult evaluateExecutionCondition(ExtensionContext extensionContext) {
+        Optional<RequiredMinKubeApiVersion> annotation = findAnnotation(extensionContext.getElement(), RequiredMinKubeApiVersion.class);
+        KubeClusterResource clusterResource = KubeClusterResource.getInstance();
+        double version = annotation.get().version();
+
+        if (Double.parseDouble(clusterResource.client().clusterKubernetesVersion()) >= version) {
+            return ConditionEvaluationResult.enabled("Test is enabled");
+        } else {
+            LOGGER.info("{} is @MultiNodeClusterOnly, but the running cluster is not multi-node cluster: Ignoring {}",
```
[the remark this answers is not in the document]
Comment (kornys) [reply systemtest/src/main/java/io/strimzi/systemtest/annotations/RequiredMinKubeApiVersionCondition.java:30]: right, will fix it

File: systemtest/src/test/java/io/strimzi/systemtest/specific/ClusterOperationST.java (hunk 1 above)
Comment (kornys) [reply systemtest/src/test/java/io/strimzi/systemtest/specific/ClusterOperationST.java:59]: weird, but I will fix it

File: systemtest/src/test/java/io/strimzi/systemtest/specific/ClusterOperationST.java (hunk 3 above)
Comment (kornys) [reply systemtest/src/test/java/io/strimzi/systemtest/specific/ClusterOperationST.java:96]: just copied :) I will fix it

**Review by scholzj — APPROVED** [review]

**Comment by scholzj** [comment]
Nice PR! 👍

**Review by Frawless — APPROVED** [review]

**Pull request merged**